jobs/common/pre-reqs.yaml

```diff
@@ -26,20 +27,104 @@
 - builder:
     name: 'install-common-tools'
     builders:
-      - shell:
+      - shell: |-
           #!/bin/bash
-          set -ex
+          set -eux
+
+          echo "Updating packages"
+          sudo DEBIAN_FRONTEND=noninteractive apt-get update
 
           echo "Installing common tools"
           sudo DEBIAN_FRONTEND=noninteractive apt-get install -y --no-install-recommends \
             make
 
+- builder:
+    name: 'install-go'
+    builders:
+    - shell:
+        !include-raw: "scripts/install-go.sh"
+
+# setup-go-environment is responsible for setting up the basic GO environment
+# needed to build and run go applications related to juju.
+#
+# This builder well set and make the GOPATH
+- builder:
+    name: 'setup-go-environment'
+    builders:
+      - install-go
+      - shell: |-
+          #!/bin/bash
+          set -eux
+
+          touch go-env.properties
+          echo "GOPATH=${WORKSPACE}/go-path" >> go-env.properties
+
+          cat go-env.properties
+      - inject:
+          properties-file: go-env.properties
+      - shell: |-
+          #!/bin/bash
+          set -ex
+          mkdir -p $GOPATH
+          rm -rf go-env.properties
+
+# set-common-environment is responsible for setting up a common environment for
+# our jobs. This shouldn't be used as a splat everything in builder that grows
+# out of control. It should be simple and only provide constructs that the vase
+# majority of jobs needs.
+#
+# It should be safe for any job to add this builder. Preferably placed at the
+# top of a builder list.
+- builder:
+    name: 'set-common-environment'
+    builders:
+      - shell: |-
+          #!/bin/bash
+          set -eux
+
+          touch build.properties
+          echo "BUILD_DIR=${WORKSPACE}/build" >> build.properties
```

Comment:
Seems like this is targetted at the jobs which compile stuff.
Is it really expected to be a common construct for other types of job?

Reply:
yeah all the integration tests use this directory.